hopsworks-common/src/main/java/io/hops/hopsworks/common/hive/HiveController.java

```diff
@@ -285,4 +290,19 @@ public Path getDbPath(String dbName) {
     return new Path(settings.getHiveWarehouse(), dbName.toLowerCase() + ".db");
   }
 
+  @TransactionAttribute(TransactionAttributeType.NOT_SUPPORTED)
+  public String getHiveServerExternalEndpoint() throws ServiceDiscoveryException {
+    return getHiveServerEndpoint(ServiceDiscoveryController.SERVICE.HIVE_SERVER_PLAIN);
+  }
+
+  @TransactionAttribute(TransactionAttributeType.NOT_SUPPORTED)
+  public String getHiveServerInternalEndpoint() throws ServiceDiscoveryException {
+    return getHiveServerEndpoint(ServiceDiscoveryController.SERVICE.HIVE_SERVER_TLS);
+  }
+
+  private String getHiveServerEndpoint(ServiceDiscoveryController.SERVICE service) throws ServiceDiscoveryException {
```

Comment:
Does this method need to be in a transaction?

Reply:
No. The `public` methods that use it are annotated as `NOT_SUPPORTED`